components/core/src/clp/ffi/ir_stream/Serializer.hpp

```diff
@@ -41,9 +41,11 @@ class Serializer {
     // Factory functions
     /**
      * Creates an IR serializer and serializes the stream's preamble.
-     * @param optional_user_defined_metadata Stream-level user-defined metadata.
+     * @param optional_user_defined_metadata Stream-level user-defined metadata, given as a JSON
+     * object.
      * @return A result containing the serializer or an error code indicating the failure:
-     * - std::errc::protocol_error on failure to serialize the preamble.
+     * - std::errc::protocol_error if failed to serialize the preamble.
```

Comment:
```suggestion
     * - std::errc::protocol_error if the stream's metadata couldn't be serialized.
```

* The grammar of the original is correct.
  * If you want to write it as "value if ..." then you can write "if the preamble could not be serialized."
* That said, I think my suggestion above is more accurate?